src/EventStore.Common/Log/ThrottledLog.cs

```diff
@@ -0,0 +1,61 @@
+using System;
+using System.Threading;
+using Serilog;
+using Serilog.Events;
+
+namespace EventStore.Common.Log; 
+
+public class ThrottledLog<T> {
+
+	private readonly ILogger _log;
+	private readonly long _duration;
+	private long _lastLogged;
+	public ThrottledLog(TimeSpan duration) {
+		_log = Serilog.Log.ForContext<T>();
+		_duration = duration.Ticks;
+		_lastLogged = DateTime.UnixEpoch.Ticks;
+	}
+
+	public bool Warning(string message) {
+		bool canLog = CanLog();
+		if (canLog) _log.Warning(message);
+		return canLog;
+	}
+	
+	public bool Fatal(string message) {
+		bool canLog = CanLog();
+		if (canLog) _log.Warning(message);
+		return canLog;
+	}
+	
+	public bool Information(string message) {
+		bool canLog = CanLog();
+		if (canLog) _log.Warning(message);
```

Comment:
```suggestion
		if (canLog) _log.Information(message);
```
The other log levels need to be changed as well

Reply:
Oops. Fixed it. My bad.